cmd/rpc/src/lib.rs

```diff
@@ -200,15 +186,63 @@ fn rpc_listen(rpc_args: &RpcArgs) -> Result<BTreeSet<Target>> {
     Ok(seen)
 }
 
-fn rpc_dump(seen: BTreeSet<Target>, image_id: &[u8]) {
-    println!(
-        "       {}         |            {}           | {}",
-        "MAC".bold(),
-        "IPv6".bold(),
-        "Compatible".bold()
+fn rpc_listen(rpc_args: &RpcArgs) -> Result<BTreeSet<Target>> {
+    // For some reason, macOS requires the interface to be non-zero:
+    // https://users.rust-lang.org/t/ipv6-upnp-multicast-for-rust-dlna-server-macos/24425
+    // https://bluejekyll.github.io/blog/posts/multicasting-in-rust/
+    let interface = match &rpc_args.interface {
+        None => {
+            if cfg!(target_os = "macos") {
+                bail!("Must specify interface with `-i` on macOS");
+            } else {
+                0
+            }
+        }
+        Some(iface) => decode_iface(iface)?,
+    };
+
+    let timeout = Duration::from_millis(rpc_args.timeout as u64);
+    let ports = [8, 8888];
+    humility::msg!(
+        "listening for {} seconds on ports {:?}...",
+        timeout.as_secs(),
+        ports
     );
 
-    println!("-------------------|---------------------------| -----------");
+    let threads = ports
+        .iter()
+        .map(|&port| {
+            std::thread::spawn(move || -> Result<BTreeSet<Target>> {
+                rpc_listen_one(timeout, interface, port)
```

Comment:
Would it be possible to ignore a permissions error here, so that you could run `humility` as an unprivileged user (albeit only being able to bind to port 8888)?  Right now I think it would just fail because 8 would fail, even though 8888 would likely then work.

Reply:
Good idea, done in fdec592ef806255411b0ddcd0ef8a0512437c724